### ISSUE TITLE
Update tests for bleeding-edge astroid namespace package changes

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#7058]
### Bug description

TestRunTC.test_regression_parallel_mode_without_filepath _______________________________________

self = <test_self.TestRunTC object at 0x10467f9d0>

    def test_regression_parallel_mode_without_filepath(self) -> None:
        # Test that parallel mode properly passes filepath
        # https://github.com/PyCQA/pylint/issues/3564
        path = join(
            HERE, "regrtest_data", "regression_missing_init_3564", "subdirectory/"
        )
>       self._test_output([path, "-j2"], expected_output="No such file or directory")

tests/test_self.py:1063: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_self.TestRunTC object at 0x10467f9d0>
args = ['--rcfile=/Users/.../pylint/pylint/testutils/testing_pylintrc', '/Users/.../pylint/tests/regrtest_data/regression_missing_init_3564/subdirectory/', '-j2']
expected_output = 'No such file or directory'

    def _test_output(self, args: list[str], expected_output: str) -> None:
        out = StringIO()
        args = _add_rcfile_default_pylintrc(args)
        self._run_pylint(args, out=out)
        actual_output = self._clean_paths(out.getvalue())
        expected_output = self._clean_paths(expected_output)
>       assert expected_output.strip() in actual_output.strip()
E       AssertionError: assert 'No such file or directory' in ''
E        +  where 'No such file or directory' = <built-in method strip of str object at 0x104692d50>()
E        +    where <built-in method strip of str object at 0x104692d50> = 'No such file or directory'.strip
E        +  and   '' = <built-in method strip of str object at 0x100bc8670>()
E        +    where <built-in method strip of str object at 0x100bc8670> = ''.strip

tests/test_self.py:170: AssertionError

TestRunTC.test_regression_recursive 

self = <test_self.TestRunTC object at 0x10465b1c0>

    def test_regression_recursive(self):
        """Tests if error is raised when linter is executed over directory not using --recursive=y"""
>       self._test_output(
            [join(HERE, "regrtest_data", "directory", "subdirectory"), "--recursive=n"],
            expected_output="No such file or directory",
        )

tests/test_self.py:1233: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_self.TestRunTC object at 0x10465b1c0>
args = ['--rcfile=/Users/.../pylint/pylint/testutils/testing_pylintrc', '/Users/.../pylint/tests/regrtest_data/directory/subdirectory', '--recursive=n']
expected_output = 'No such file or directory'

    def _test_output(self, args: list[str], expected_output: str) -> None:
        out = StringIO()
        args = _add_rcfile_default_pylintrc(args)
        self._run_pylint(args, out=out)
        actual_output = self._clean_paths(out.getvalue())
        expected_output = self._clean_paths(expected_output)
>       assert expected_output.strip() in actual_output.strip()
E       AssertionError: assert 'No such file or directory' in ''
E        +  where 'No such file or directory' = <built-in method strip of str object at 0x104692d50>()
E        +    where <built-in method strip of str object at 0x104692d50> = 'No such file or directory'.strip
E        +  and   '' = <built-in method strip of str object at 0x100bc8670>()
E        +    where <built-in method strip of str object at 0x100bc8670> = ''.strip

tests/test_self.py:170: AssertionError

TestRunTC.test_regression_recursive_current_dir

self = <test_self.TestRunTC object at 0x10465b2b0>

    def test_regression_recursive_current_dir(self):
        with _test_sys_path():
            # pytest is including directory HERE/regrtest_data to sys.path which causes
            # astroid to believe that directory is a package.
            sys.path = [
                path
                for path in sys.path
                if not os.path.basename(path) == "regrtest_data"
            ]
            with _test_cwd():
                os.chdir(join(HERE, "regrtest_data", "directory"))
                breakpoint()
>               self._test_output(
                    ["."],
                    expected_output="No such file or directory",
                )

tests/test_self.py:1345: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_self.TestRunTC object at 0x10465b2b0>, args = ['--rcfile=/Users/.../pylint/pylint/testutils/testing_pylintrc', '.']
expected_output = 'No such file or directory'

    def _test_output(self, args: list[str], expected_output: str) -> None:
        out = StringIO()
        args = _add_rcfile_default_pylintrc(args)
        self._run_pylint(args, out=out)
        actual_output = self._clean_paths(out.getvalue())
        expected_output = self._clean_paths(expected_output)
>       assert expected_output.strip() in actual_output.strip()
E       AssertionError: assert 'No such file or directory' in ''
E        +  where 'No such file or directory' = <built-in method strip of str object at 0x104692d50>()
E        +    where <built-in method strip of str object at 0x104692d50> = 'No such file or directory'.strip
E        +  and   '' = <built-in method strip of str object at 0x100bc8670>()
E        +    where <built-in method strip of str object at 0x100bc8670> = ''.strip

tests/test_self.py:170: AssertionError
======================================================= short test summary info ========================================================
FAILED tests/test_self.py::TestRunTC::test_regression_parallel_mode_without_filepath - AssertionError: assert 'No such file or direct...
FAILED tests/test_self.py::TestRunTC::test_regression_recursive - AssertionError: assert 'No such file or directory' in ''
FAILED tests/test_self.py::TestRunTC::test_regression_recursive_current_dir - AssertionError: assert 'No such file or directory' in ''
================================================== 3 failed, 121 deselected in 44.75s ==================================================

### Command used

```shell
pytest tests/test_self.py
```

### Pylint output

```shell
see above
```

### Expected behavior

n/a

### Pylint version

```shell
pylint 2.15.0-dev0
astroid 2.12.0-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

macOS

**Comment by DanielNoord** [comment]
Those have been happening for some weeks. Should have probably reported but thought it was something to do with my own environment.

**Comment by jacobtylerwalls** [comment]
I've been ignoring it also. It's bleeding-edge astroid. I guess we can close since we'll just deal with later. But at a glance, the failures look better: instead of saying we can't lint a package with a missing `__init__.py`, we lint it.
